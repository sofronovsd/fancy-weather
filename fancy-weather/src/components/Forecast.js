import React from "react";
import Moment from "react-moment";
import weatherMap from "../utils/weatherMap";

export default function Forecast({language, forecast, isC}) {
    return (
        <div className="forecast-container">
            {forecast.map((day, index) => {
                return <div className="forecast-container__item" key={index}>
                    <img
                        className="icon icon_secondary"
                        src={day.weather_code && weatherMap[day.weather_code.value]}
                        alt="weather-icon"
                    />
                    <p className="text text__medium">
                        <Moment
                            locale={language}
                            date={day.observation_time.value}
                            format="dddd"
                        />
                    </p>
                    <p className="text">
                        {`${isC ? Math.floor((day.temp[0].min.value + day.temp[1].max.value) / 2) :
                            Math.floor((((day.temp[0].min.value + day.temp[1].max.value) / 2) * 9) / 5 + 32)}`}
                        <sup>&deg;{isC ? 'C' : 'F'}</sup>
                    </p>
                </div>
            })}
        </div>
    )
}
