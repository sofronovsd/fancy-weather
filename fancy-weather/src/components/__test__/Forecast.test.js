import Forecast from "../Forecast";
import React from "react";
import ReactDOM from "react-dom";
import {render} from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom"

describe('Test forecast functionality', () => {

  const forecast=[
    {
      weather_code: "cloudy",
      observation_time: {
        value: 10
      },
      temp: [
        {
          min: {
            value: 10
          }
        },
        {
          max: {
            value: 10
          }
        }
      ]
    }
  ];

  it('render forecast without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Forecast forecast={forecast}/>, div)
  });

  it('render forecast correctly', () => {

    const {getByTestId} = render(<Forecast forecast={forecast}/>);
    expect(getByTestId('forecastContainer')).toBeDefined();
    expect(getByTestId('forecastItem0')).toBeDefined();
  });

  it('forecast matches snapshot', () => {
    const tree = renderer.create(<Forecast forecast={forecast}/>);
    expect(tree).toMatchSnapshot();
  });
});
