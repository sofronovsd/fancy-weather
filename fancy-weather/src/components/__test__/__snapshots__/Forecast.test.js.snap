// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test forecast functionality forecast matches snapshot 1`] = `
<div
  className="forecast-container"
  data-testid="forecastContainer"
>
  <div
    className="forecast-container__item"
    data-testid="forecastItem0"
  >
    <img
      alt="weather-icon"
      className="icon icon_secondary"
    />
    <p
      className="text text__medium"
    >
      <time
        dateTime={"1970-01-01T00:00:00.010Z"}
      >
        Thursday
      </time>
    </p>
    <p
      className="text"
    >
      50
      <sup>
        Â°
        F
      </sup>
    </p>
  </div>
</div>
`;
